### PR TITLE
CMR-6222: As a CMR Developer, I Want to Run Dev-system on Top of ES 7.5.2

### PR DESCRIPTION
**Pull request by ygliuvt**
This PR runs dev-system against the external Elasticsearch 7.5.2 that is running locally via docker. The ES can be started by running `docker-compose up` in the `resources/docker` directory under `elastic-utils-lib`. The dev-system can be started successfully and most of the search integration tests can also run successfully with the code changes in this PR. 

I tried `https://github.com/spandex-project/spandex`, but found it too low level and is too big a departure from our existing code base. On the other end, the Elastisch 5.0.0-beta1 release fixed most of upgrade issues for us and aligns well with our existing code base. So I decided to continue using the Elastisch library and add in fixes in the helper namespaces in `elastic-utils-lib` when necessary.

This PR will not run in CI because CI still depends on embedded ES which is covered in CMR-6133, so don't be alarmed if you see build failure on this branch.

### CODE REVIEW

File: elastic-utils-lib/resources/docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+version: '3'
+services:
+
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch:7.5.2
+    container_name: cmr-es-elastic
+    environment:
+      - http.host=0.0.0.0
+      - transport.host=127.0.0.1
+      - ELASTIC_PASSWORD=kibana
```
Comment (daniel-zamora) [inline elastic-utils-lib/resources/docker/docker-compose.yml]: kibana? weird password.

File: elastic-utils-lib/project.clj [hunk 2]
```diff
@@ -8,7 +8,7 @@
                [potemkin]]
   :dependencies [[cheshire "5.8.1"]
                  [clj-http "2.3.0"]
-                 [clojurewerkz/elastisch "2.2.2"]
+                 [clojurewerkz/elastisch "5.0.0-beta1"]
```
Comment (daniel-zamora) [inline elastic-utils-lib/project.clj]: I'm a little wary of using an unstable library that has a history of extremely slow updates for something this important.  We are potentially trading reliability for expedience, but I wasn't a huge fan of using spandex either.  If this works then great, it just makes me nervous.

Comment (ygliuvt) [reply elastic-utils-lib/project.clj]: We always have the option to roll our own. This makes it more convenient now that we have ruled out Spandex.

**Comment by sxu123** [comment]
@ygliuvt, What does "most of the search integration tests can also run successfully with the code changes in this PR." imply?  So once this is merged, we are expecting some tests failures locally?

File: elastic-utils-lib/src/cmr/elastic_utils/es_helper.clj [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  [conn index mapping-type opts]
+  (doc/search conn index mapping-type opts))
+
+(defn count-query
+  "Performs a count query over one or more indexes and types"
+  [conn index mapping-type query]
+  (doc/count conn index mapping-type query))
+
+(defn scroll
+  "Performs a count query over one or more indexes and types"
```
Comment (sxu123) [inline elastic-utils-lib/src/cmr/elastic_utils/es_helper.clj]: doc string needs to be changed.

File: elastic-utils-lib/resources/docker/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+    volumes:
+      - ${PWD}/data/elastic:/usr/share/elasticsearch/data
+    ports:
+      - 9209:9200
+    networks:
+      - cmr-es-net
```
Comment (sxu123) [inline elastic-utils-lib/resources/docker/docker-compose.yml]: where is this network defined?

File: elastic-utils-lib/src/cmr/elastic_utils/es_helper.clj (hunk 3 above)
Comment (sxu123) [reply elastic-utils-lib/src/cmr/elastic_utils/es_helper.clj]: Where is this es_helper.clj used?  I can't seem to find it.

File: elastic-utils-lib/resources/docker/docker-compose.yml (hunk 4 above)
Comment (ygliuvt) [reply elastic-utils-lib/resources/docker/docker-compose.yml]: last line of the file

File: elastic-utils-lib/src/cmr/elastic_utils/es_helper.clj (hunk 3 above)
Comment (ygliuvt) [reply elastic-utils-lib/src/cmr/elastic_utils/es_helper.clj]: Done.

File: elastic-utils-lib/resources/docker/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      - xpack.security.enabled=false
+      - xpack.monitoring.enabled=false
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+    volumes:
+      - ${PWD}/data/elastic:/usr/share/elasticsearch/data
+    ports:
+      - 9209:9200
```
Comment (CayvonH) [inline elastic-utils-lib/resources/docker/docker-compose.yml]: Any reason for 9209?

Comment (ygliuvt) [reply elastic-utils-lib/resources/docker/docker-compose.yml]: It is the port that is configured for default external elasticsearch port in dev-system. I use this port number to make it easy to run with dev-system where we only need to switch the elasticsearch to :external mode.

**Comment by ygliuvt** [comment]
@sxu123 This PR is the starting point of the update ES feature epic. As I have explained in the description of the PR, the build will fail because 1) we don't have ES 7.5.2 running in embedded mode which is used by CI, and 2) we only added 7.5.2 support to the basic index creation/deletion apis of ES 7.5.2. The support for the other apis that CMR uses will be added as we work through the rest of the tickets in the epic. Only until the end of the epic will we see all the tests passing in CI.

File: elastic-utils-lib/resources/docker/docker-compose.yml (hunk 1 above)
Comment (ygliuvt) [reply elastic-utils-lib/resources/docker/docker-compose.yml]: This is an internal password that nobody cares. :)

**Pull request merged**